packages/connect-web/src/impl/core-in-popup.ts

```diff
@@ -0,0 +1,224 @@
+import EventEmitter from 'events';
+
+// NOTE: @trezor/connect part is intentionally not imported from the index so we do include the whole library.
+import {
+    POPUP,
+    IFRAME,
+    UI_EVENT,
+    createErrorMessage,
+    UiResponseEvent,
+    CallMethodPayload,
+} from '@trezor/connect/src/events';
+import * as ERRORS from '@trezor/connect/src/constants/errors';
+import type { ConnectSettings, Manifest, Response } from '@trezor/connect/src/types';
+import { ConnectFactoryDependencies, factory } from '@trezor/connect/src/factory';
+import { initLog, setLogWriter, LogMessage, LogWriter, Log } from '@trezor/connect/src/utils/debug';
+import * as popup from '../popup';
+
+import { parseConnectSettings } from '../connectSettings';
+import { Login } from '@trezor/connect/src/types/api/requestLogin';
+
+/**
+ * Base class for CoreInPopup methods for TrezorConnect factory.
+ * This implementation is directly used here in connect-web, but it is also extended in connect-webextension.
+ */
+export class CoreInPopup implements ConnectFactoryDependencies {
+    public eventEmitter = new EventEmitter();
+    protected _settings: ConnectSettings;
+
+    /**
+     * setup logger.
+     * service worker cant communicate directly with sharedworker logger so the communication is as follows:
```

Comment:
Just for my clarification.
This is not possible for the main reason that service worker runs in the context of the webextension and sharedworker is in trezor.io domain so they need a connection where there was content script injected in order to allow communication between those two domains. 
But you suggested that we could also inject it into log.html so service worker would be able to talk directly with it.

Right?

Reply:
Actually I just moved this code from the webextension package, along with this comment. 
It seems this was originally commented by @mroz22 
But the comment is only relevant to the webextension, not `connect-web` so I should move it back again.